### PR TITLE
Support Node 18

### PR DESCRIPTION
**Pull request by BlackthornYugen**
This change is [Reviewable](https://reviewable.io/reviews/mongo-express/mongo-express/851)
---

## ✅ Node 18

It looks like node-18 works without any changes

## ⚠️ Updated Dependency
~~`npm audit fix --force` to update `express-fileupload`, before I take off "draft" I will make sure that file upload still works.~~

**edit**  I've tested that export/import still works with the change

## ~~⚠️ Failing Tests~~

~~28/58 tests fail at the moment on my machine. I am having trouble getting any of the router tests to run:~~

image

**edit**: it seems to work on the CI build

### CODE REVIEW

**Review by dozoisch — COMMENTED** [review]
> Reviewed 7 of 9 files at r1.
> *[Reviewable](https://reviewable.io/reviews/mongo-express/mongo-express/851)* status: 4 of 9 files reviewed, 1 unresolved discussion (waiting on @BlackthornYugen and @dozoisch)

___
*[`lib/scripts/database.js` line 67 at r1](https://reviewable.io/reviews/mongo-express/mongo-express/851#-N2KGSAYCC_hjOA6Kd6I:-N2KGSAYCC_hjOA6Kd6J:b97b6zn) ([raw file](https://github.com/mongo-express/mongo-express/blob/f088747d45d2b28f158dabb0e910343b6c684432/lib/scripts/database.js#L67)):*
> ```JavaScript
>     })
>       .done(function (res) {
>         console.log(res);
> ```

minor: just making sure that those changes are done on purpose because that is a big change of behavior... In all honesty, I dont remember what shows up in the alert but wanted to point it out

**Comment by BlackthornYugen** [comment]
> making sure that those changes are done on purpose because that is a big change of behavior...

Good point, maybe for now I can just suppress the lint error? or it's just a lint warning and it isn't failing the build, I could also just revert those changes too.

**Comment by BlackthornYugen** [comment]
@dozoisch it might be nicer if the lint changes could be moved into their own PR so I created #852. I guess I broke the build with the PEM change in #815.

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "csurf": "1.10.0",
     "errorhandler": "1.5.1",
     "express": "4.17.1",
-    "express-fileupload": "^0.4.0",
+    "express-fileupload": "^1.3.1",
```
Comment (BlackthornYugen) [inline package.json:41]: it probably makes sense to have this in it's own PR too. I like how the other version bumps are in a commit named "bump x from y to z" with a "dependencies" tag so it is easy to spot these in the git history and to update history.md. 

Is #592 the only dependabot PR for fileupload? It seems to be updating more than just fileupload.

**Comment by rtritto** [comment]
Node v12 is no longer maintained.
It must be removed because some new dependencies add support to Node v18 and remove to Node v12 in same release.

**Comment by BlackthornYugen** [comment]
Replaced by #911 #916 and #914

**Pull request closed without merging**